tools/provenance-generator/pkg/parse.go

```diff
@@ -0,0 +1,169 @@
+	// `https://github.com/org/repo.git`.
+	CloneURI string `json:"clone_uri,omitempty"`
+	// SkipSubmodules determines if submodules should be
+	// cloned when the job is run. Defaults to false.
+	SkipSubmodules bool `json:"skip_submodules,omitempty"`
+	// CloneDepth is the depth of the clone that will be used.
+	// A depth of zero will do a full clone.
+	CloneDepth int `json:"clone_depth,omitempty"`
+	// SkipFetchHead tells prow to avoid a git fetch <remote> call.
+	// Multiheaded repos may need to not make this call.
+	// The git fetch <remote> <BaseRef> call occurs regardless.
+	SkipFetchHead bool `json:"skip_fetch_head,omitempty"`
+}
+
+type Record struct {
+	Refs     Refs      `json:"refs"`
+	Commands []Command `json:"commands,omitempty"`
+	Failed   bool      `json:"failed,omitempty"`
+
+	// FinalSHA is the SHA from ultimate state of a cloned ref
+	// This is used to populate RepoCommit in started.json properly
+	FinalSHA string `json:"final_sha,omitempty"`
+}
+
+// Can't seem to import "k8s.io/test-infra/prow/pod-utils/clone" library :(
+type Command struct {
+	Command string `json:"command"`
+	Output  string `json:"output,omitempty"`
+	Error   string `json:"error,omitempty"`
+}
+
+func LoadParameters(config Config) Config {
+	// Parse Entrypoint of the prowjob to pull args, path, etc
+	entryPointOpsRaw := os.Getenv(config.EntryPointOptsVariable)
+	entryPointOpts := &entrypoint.Options{}
+	err := json.Unmarshal([]byte(entryPointOpsRaw), entryPointOpts)
+	if err != nil {
+		log.Fatalf("failed to unmarshal ENTRYPOINTS_OPTIONS env variable %v", err)
+	}
+	config.EntryPointOpts = entryPointOpts
+
+	// Sanity checks
+	if config.EntryPointOpts.Args[0] != "runner.sh" {
+		log.Fatal("this prowjob is misconfigured, expecting runner.sh to be called first")
+	}
+
+	config.Arguments = map[string]string{}
+	keys := []string{}
+
+	// Parse entrypoint flags
+	for _, elem := range config.EntryPointOpts.Args {
+		params := strings.Split(elem, "=") // expected values --something=foo
+		if len(params) == 1 {
+			config.Arguments[params[0]] = ""
+			keys = append(keys, params[0])
+		} else if len(params) == 2 {
+			config.Arguments[params[0]] = params[1]
+			keys = append(keys, params[0])
+		}
+	}
+	config.ArgumentsInsertOrder = keys
+
+	//Appoximate start time of job
+	fileInfo, err := os.Stat(config.CloneLogPath)
+	if err != nil {
+		log.Fatalf("failed to open file %v", err)
+	}
+	config.StartTime = fileInfo.ModTime()
+
+	byteValue, _ := os.ReadFile(config.CloneLogPath)
+	cloneLogs := []Record{}
+
+	err = json.Unmarshal(byteValue, &cloneLogs)
+	if err != nil {
+		log.Fatalf("failed to unmarshal clone-logs file %v", err)
+	}
+	config.CloneRecords = cloneLogs
+
+	// Lets strip out the empty refs that is inserted by prow
+	for i := len(config.CloneRecords) - 1; i >= 0; i-- {
+		cloneRecord := config.CloneRecords[i]
+		if cloneRecord.FinalSHA == "" {
+			config.CloneRecords = append(config.CloneRecords[:i],
+				config.CloneRecords[i+1:]...)
+		}
+	}
```

Comment:
It's probably slightly more efficient to keep two counters and trim the slice at the end, but it may not be important:

```golang
for i, j := 0; i < len(config.CloneRecords) -1; i++ {
  if config.CloneRecords[i].FinalSHA == "" {
    continue
  }
  if i != j {
    config.CloneRecords[j] = config.CloneRecords[i]
  }
  j++
}
config.CloneRecords = config.CloneRecords[:j-1]
```